src/plugins/data_views/public/data_views/data_views_api_client.ts

```diff
@@ -6,33 +6,61 @@
  * Side Public License, v 1.
  */
 
-import { HttpSetup } from '@kbn/core/public';
+import { HttpSetup, HttpResponse } from '@kbn/core/public';
 import { DataViewMissingIndices } from '../../common/lib';
 import { GetFieldsOptions, IDataViewsApiClient } from '../../common';
 import { FieldsForWildcardResponse } from '../../common/types';
-import { FIELDS_FOR_WILDCARD_PATH } from '../../common/constants';
+import { FIELDS_FOR_WILDCARD_PATH, FIELDS_PATH } from '../../common/constants';
 
 const API_BASE_URL: string = `/api/index_patterns/`;
 const version = '1';
 
+async function sha1(str: string) {
+  const enc = new TextEncoder();
+  const hash = await crypto.subtle.digest('SHA-1', enc.encode(str));
+  return Array.from(new Uint8Array(hash))
+    .map((v) => v.toString(16).padStart(2, '0'))
+    .join('');
+}
+
 /**
  * Data Views API Client - client implementation
  */
 export class DataViewsApiClient implements IDataViewsApiClient {
   private http: HttpSetup;
+  private getCurrentUserId: () => Promise<string | undefined>;
 
   /**
    * constructor
    * @param http http dependency
    */
-  constructor(http: HttpSetup) {
+  constructor(http: HttpSetup, getCurrentUserId: () => Promise<string | undefined>) {
     this.http = http;
+    this.getCurrentUserId = getCurrentUserId;
```

Comment:
It looks like this constructor is called before the `userIdGetter` is set (inside src/plugins/data_views/public/plugin.ts), which means this will always return `undefined`.

Reply:
fixed, good catch